File: code/appd-9.erl
P = {ok,{note,""}}.
{ok,_} = P. % -> {ok,{note,[]}}
{Ret,_} = P. % -> {ok,{note,[]}}
Ret. % -> ok
[] = P. % -> exception error: no match of right hand side value {ok,{note,[]}}

File: code/appd-9.ex
p = {:ok,{:note,""}}
{:ok,_} = p # -> {:ok, {:note, ""}}
{ret,_} = p # ->{:ok, {:note, ""}}
ret # -> :ok
[] = p # -> (MatchError) no match of right hand side value: {:ok, {:note, ""}}
